### PR TITLE
Add unit tests for org.springframework.security.oauth2.common.exceptions.OAuth2Exception

### PR DESCRIPTION
**Pull request by ThomasPerkins1123**
Hi,

I've analysed your codebase and noticed that `org.springframework.security.oauth2.common.exceptions.OAuth2Exception` is not fully tested.

I've written some tests for the methods in this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests will help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other classes that you consider important in a subsequent PR.

### CODE REVIEW

**Review by JohnLBergqvist — CHANGES_REQUESTED** [review]
You've used `test` in the method names twice.

**Comment by ThomasPerkins1123** [comment]
> You've used `test` in the method names twice.

Sorry I did fix this but forgot to push! Is fixed now

File: spring-security-oauth2/src/test/java/org/springframework/security/oauth2/common/exceptions/OAuth2ExceptionTest.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  }
+
+  @Test
+  public void testCreate3() {
+    OAuth2Exception oAuth2Exception = OAuth2Exception.create("unsupported_grant_type", null);
+
+    Assert.assertNull(oAuth2Exception.getAdditionalInformation());
+    Assert.assertEquals(oAuth2Exception.getClass(), UnsupportedGrantTypeException.class);
+    Assert.assertEquals("unsupported_grant_type",
+                        oAuth2Exception.getMessage());
```
Comment (JohnLBergqvist) [inline spring-security-oauth2/src/test/java/org/springframework/security/oauth2/common/exceptions/OAuth2ExceptionTest.java]: Does this need to be on a 2nd line?

File: spring-security-oauth2/src/test/java/org/springframework/security/oauth2/common/exceptions/OAuth2ExceptionTest.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  }
+
+  @Test
+  public void testCreate10() {
+    OAuth2Exception oAuth2Exception = OAuth2Exception.create("unsupported_response_type", null);
+
+    Assert.assertNull(oAuth2Exception.getAdditionalInformation());
+    Assert.assertEquals(oAuth2Exception.getClass(), UnsupportedResponseTypeException.class);
+    Assert.assertEquals("unsupported_response_type",
+                        oAuth2Exception.getMessage());
```
Comment (JohnLBergqvist) [inline spring-security-oauth2/src/test/java/org/springframework/security/oauth2/common/exceptions/OAuth2ExceptionTest.java]: Same as above

File: spring-security-oauth2/src/test/java/org/springframework/security/oauth2/common/exceptions/OAuth2ExceptionTest.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+  }
+
+  @Test
+  public void testCreate11() {
+    OAuth2Exception oAuth2Exception = OAuth2Exception.create("redirect_uri_mismatch", null);
+
+    Assert.assertNull(oAuth2Exception.getAdditionalInformation());
+    Assert.assertEquals(oAuth2Exception.getClass(), RedirectMismatchException.class);
+    Assert.assertEquals("redirect_uri_mismatch",
+                        oAuth2Exception.getMessage());
```
Comment (JohnLBergqvist) [inline spring-security-oauth2/src/test/java/org/springframework/security/oauth2/common/exceptions/OAuth2ExceptionTest.java]: Same as above

**Review by JohnLBergqvist — CHANGES_REQUESTED** [review]

**Review by JohnLBergqvist — APPROVED** [review]

**Review by petersear — APPROVED** [review]

**Pull request closed without merging**